Description: Enable UPnP usage in package
Author: r4sas <r4sas@i2pmail.org>

Reviewed-By: r4sas <r4sas@i2pmail.org>
Last-Update: 2024-12-30

--- i2pd.orig/Makefile
+++ i2pd/Makefile
@@ -31,7 +31,7 @@ # import source files lists
 include filelist.mk
 
 USE_STATIC      := $(or $(USE_STATIC),no)
-USE_UPNP        := $(or $(USE_UPNP),no)
+USE_UPNP        := $(or $(USE_UPNP),yes)
 DEBUG           := $(or $(DEBUG),yes)
 
 # for debugging purposes only, when commit hash needed in trunk builds in i2pd version string
